nibabel/streamlines/compact_list.py

```diff
@@ -0,0 +1,232 @@
+import numpy as np
+
+
+class CompactList(object):
```

Comment:
Can you think of a better name?   `CompactList` seems a bit generic for what this is, which is (I believe) a sequence of 2D arrays of shape (N, M) where M must be the same for every array.

How about something like ArraySequence ?